.github/workflows/pipeline.yml

```diff
@@ -0,0 +1,160 @@
+          echo "##[set-output name=path;]$THE_ESY__CACHE_INSTALL_PATH"
+      - name: Restore esy cache
+        uses: actions/cache@v1
+        with:
+          path: ${{ steps.esy_cache_path.outputs.path }}
+          key: v1-esy-${{ matrix.os }}-${{ hashFiles('**/index.json') }}
+          restore-keys: |
+            v1-esy-${{ matrix.os }}-
+
+      - name: Install and build
+        run: |
+          esy install 
+          esy dune build --root . --only-packages '#{self.name}' --ignore-promoted-rules --no-config --profile release-static
+      - name: (only on release) Upload artifacts ${{ matrix.os }}
+        if: github.event_name != 'pull_request' && startsWith(github.ref, 'refs/tags/')
+        uses: actions/upload-artifact@master
+        with:
+          name: ${{ matrix.os }}
+          path: _build/default/bin/bin.exe
+
+  test_and_build:
+    name: ${{ matrix.os }} / node-${{ matrix.node-version }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        node-version: [12.x]
+        os: [windows-latest, macOS-latest]
+
+    steps:
+      - uses: actions/checkout@v1
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Install esy
+        run: |
+          npm install -g esy@latest
+
+      - name: Get esy store path
+        id: esy_cache_path
+        shell: bash
+        run: |
+          # COMPUTE THE ESY INSTALL CACHE LOCATION AHEAD OF TIME
+          if [ "${{ matrix.os }}" == "windows-latest" ]; then
+            THE_ESY__CACHE_INSTALL_PATH=$HOME/.esy/3_/i
+            THE_ESY__CACHE_INSTALL_PATH=$( cygpath --mixed --absolute "$THE_ESY__CACHE_INSTALL_PATH")
+          else
+            DESIRED_LEN="86"
+            HOME_ESY3="$HOME/.esy/3"
+            HOME_ESY3_LEN=${#HOME_ESY3}
+            NUM_UNDERS=$(echo "$(($DESIRED_LEN-$HOME_ESY3_LEN))")
+            UNDERS=$(printf "%-${NUM_UNDERS}s" "_")
+            UNDERS="${UNDERS// /_}"
+            THE_ESY__CACHE_INSTALL_PATH=${HOME_ESY3}${UNDERS}/i  
+          fi
+          echo "THE_ESY__CACHE_INSTALL_PATH: $THE_ESY__CACHE_INSTALL_PATH"
+          echo "##[set-output name=path;]$THE_ESY__CACHE_INSTALL_PATH"
+      - name: Restore esy cache
+        uses: actions/cache@v1
+        with:
+          path: ${{ steps.esy_cache_path.outputs.path }}
+          key: v1-esy-${{ matrix.os }}-${{ hashFiles('**/index.json') }}
+          restore-keys: |
+            v1-esy-${{ matrix.os }}-
+
+      - name: Install and build
+        run: |
+          esy install 
+          esy b
+      - name: (only on release) Upload artifacts ${{ matrix.os }}
+        if: github.event_name != 'pull_request' && startsWith(github.ref, 'refs/tags/')
+        uses: actions/upload-artifact@master
+        with:
+          name: ${{ matrix.os }}
+          path: _build/default/bin/bin.exe
+
+  publish:
+    needs: test_and_build
```

Comment:
Shouldn't this also need `test_and_build_linux`?

Reply:
Probably yes but in reality windows build is the slowest. I haven't had any issues with it in `graphql_ppx_re` so far 🙂